### ISSUE TITLE
Allow to define default value and range for AudioPorts used as CV ports

### ISSUE THREAD

**Issue by SpotlightKid** [DISTRHO/DPF#237]
Augment `AudioPort` struct to allow to set default, minimum and maximum value and add new hint to indicate that DPF should export appropriate port properties or meta data for these ports.

Maybe also allow to set the the `isLogarithmic` hint on `AudioPort.hint`.

## More information

Log of IRC discussion, which lead to this ticket:

```
[23:39:14] <strogon14> falktx: ok, thanks for the info. So would it make sense to support CV ports in DPF? Currently, I believe there is no way to set the name of audio ports and add the neccessary port properties in the TTL file, is there? Or is this too LV2 specific for DPF?
[23:51:21] <@falktx> strogon14: you can do all that already. see https://distrho.github.io/DPF/classPlugin.html#a75f30935198a0acba25f4ea92af72dc4 and https://distrho.github.io/DPF/structAudioPort.html
[23:52:13] <@falktx> just reimplement the function to set the name and hits. on the hints, use kAudioPortIsCV
[23:54:23] <strogon14> falktx: allright! So, I guess in VST and LADSPA formats these ports will only be distinguishable from normal audio ports by name?
[23:54:54] <@falktx> strogon14: I guess.. I just simply recommend not exporting for those formats
[23:55:27] <@falktx> not sure if the cv hint triggers jack meta data setup, easy to add if not
[23:55:53] <strogon14> I guess as I only have CV input ports, they can do no great harm.
[23:56:07] <strogon14> * as long as
[23:57:09] <strogon14> CV ports would make jack format interesting again for plugins without custom UI.
[00:04:27] <strogon14> falktx: there is currently no way to add lv2:default, lv2:minimum and lv2:maximum to CV ports, though, is there?
[00:06:29] <@falktx> no
[00:07:06] <@falktx> but the API can always be extended of course
[00:09:28] <alextee[m]> that would solve many problems
[00:09:33] <alextee[m]> adding default/min/max on CV
[00:09:36] <alextee[m]> can be easily converted
[00:10:00] <strogon14> The LV2 spec says about CVPorts, "In particular, this port type does not imply any range, unit, or meaning for its values. However, if there is no inherent unit to the values, for example if the port is used to modulate some other value, then plugins SHOULD use a normalized range, either from -1.0 to 1.0, or from 0.0 to 1.0."
[00:10:51] <strogon14> I'm not suert how to inpterpret that. Is it ok to add min/max/default to CVPorts or not? And how do (would) LV2 host handle the default value?
[00:11:02] <strogon14> s/suert/sure/
[00:12:25] <@falktx> you can add ranges and unit
[00:12:31] <@falktx> that would give them meaning
[00:12:57] <@falktx> this is one of the reasons why http://moddevices.com/ns/mod/#CVPort exists
[00:15:03] <strogon14> So would you propose adding a Parameter member to the AudioPort struct? Or overload/change the signature of initAudioPort to pass in a Parameter reference?
[00:15:34] <@falktx> nah, just have the def,min,max in the audioport struct
[00:15:53] <@falktx> I looked into it now, the parameterRange struct has functions which are of no use for audio
[00:16:00] <@falktx> so it would get confusing
[00:16:33] <@falktx> I should add mod:volts as unit on those ports, to make them a bit more meaningful
[00:17:03] <strogon14> would integer and logarithmic CVPorts make sense, theoretically?
[00:17:09] <@falktx> strogon14: I propose the following. new hint for audio ports, to tell that cvport has ranges defined
[00:18:00] <@falktx> when that is set, dpf writes the ranges
[00:18:00] <@falktx> and output ports should not have default btw
[00:18:03] <@falktx> strogon14: IMO integer makes no sense for cv
[00:18:33] <@falktx> is there any real CV hw that only deals with integers? I think that would be very odd
[00:18:46] <milkii> when a parameter is exposed as a cv port
[00:19:15] <@falktx> maybe for its output, some cv modules perhaps output only in "integer" fashion, but receiving too?
[00:19:21] <strogon14> falktx: yeah, probably. they whole point is that in the real world they have no discrete values
[00:20:25] <@falktx> milkii: still seems a bit of a stretch. and the host will just round down the received values
[00:21:45] <nilshi> 0 to MAXINT should be smooth enough
[00:21:46] <milkii> more as a description?
[00:21:53] <strogon14> falktx: a new hint for AudioPorts sounds good and easy to use.
[00:22:26] <@falktx> ranges for cv ports do not always make sense, so should be optional yes
[00:22:35] <@falktx> strogon14: care to open a ticket about this?
[00:22:46] <strogon14> falktx: ok
```

**Comment by falkTX** [comment]
Coming back to this.. I am now against allowing arbitrary CV ranges.
I dealt with this for MOD, arbitrary ranges are very problematic for connecting things together, it is best if everything follows from the same standard way of doing things.

So I propose the following:
- add flags to allow to set CV range as "unipolar+", "bipolar" and "unipolar-"
- add flag to scale CV range (ie, use 10v or -5/+5v as MOD)

So basically when range flag is set, it will be by default either -1.0 <-> +1.0 (unipolar), or 0 <-> 1.0 (unipolar+) or -1.0 <-> 0.0 (unipolar-).
This is kinda what ams-lv2 is doing already.
If scale CV flag is set, port will use "real cv value" alike what MOD does.

If no range CV flag is set, no ranges are set in the LV2 TTL.

**Comment by SpotlightKid** [comment]
That sounds reasonable to me.

**Comment by falkTX** [comment]
Done in ead50d1c850071a4db3883b692246f8bce46dd15
